packages/ui/src/components/editor/panels/Hierarchy/container/index.tsx

```diff
@@ -145,11 +145,16 @@ function HierarchyPanelContents(props: { sceneURL: string; rootEntityUUID: Entit
 
   const searchedNodes: HierarchyTreeNodeType[] = []
   if (searchHierarchy.value.length > 0) {
-    const condition = new RegExp(searchHierarchy.value.toLowerCase())
-    entityHierarchy.value.forEach((node) => {
-      if (node.entity && condition.test(getComponent(node.entity, NameComponent)?.toLowerCase() ?? ''))
-        searchedNodes.push(node)
-    })
+    try {
+      const adjustedSearchValue = searchHierarchy.value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
```

Comment:
Can you add this regex to this module: `packages/common/src/regex/index.ts`?

And few test cases here: `packages/common/tests/regex.test.ts` (ChatGPT can generate them)

Reply:
done